cosmos.md

```diff
@@ -0,0 +1,198 @@
+title: Cosmos Namespace
+author: Simon Warta (@webmaster128)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/issues/5, https://github.com/ChainAgnostic/CAIPs/issues/6, https://github.com/ChainAgnostic/CAIPs/pull/1
+status: Draft
+type: Standard
+created: 2022-03-01
+updated: 2022-03-01
+requires: CAIP-2, CAIP-10
+supercedes: CAIP-5
+---
+
+# Namespace for Cosmos chains
+
+This document describes the syntax and structure of the [Cosmos][] namespace,
+which formalized a URI scheme for references on the broader Cosmos ecosystem.
+This ecosystem includes many different kinds of blockchains, variously
+interoperable and interconnected, but share the consensus rules and
+[configuration objects][] of the Tendermint consensus engine.  The namespace thus
+defined includes blockchains generated using the [Cosmos
+SDK](https://github.com/cosmos/cosmos-sdk) such as Cosmoshub, Binance, and the
+Cosmos Testnets, but also [Weave](https://github.com/iov-one/weave)-based
+blockchains, like IOV.
+
+## CAIP-2
+
+*For context, see the [CAIP-2][] specification.*
+
+### Reference basics
+
+Cosmos chains consist of the namespace prefix `cosmos` and a reference.
+
+Each reference is a `chain_id` from the genesis file used to initiate a
+Tendermint blockchain; this file is a JSON object, and thus the `chain_id` is a
+unique, JSON-compatible unicode string. For an example, see the [configuration
+objects][] section of the tendermint core specification.  It can be referenced
+directly or by its hash according to an algorithm defined below.  All nodes of
+any given network also return the string when queried for basic configuration
+information, as described below. 
+
+Note: An empty or null `chain_id` must be treated as an error.
+
+### Direct References
+
+If the `chain_id` matches the case-sensitive pattern `[-a-zA-Z0-9]{1,32}` and
+does not start with "hashed-", it can be assumed to refer to a `chain_id`.  If
+the first 7 characters are `hashed-`, however, it should be assumed to be a hash
+of a `chain_id`; the original can be retrieved from any node as described below.  
+
+Note: some `chain_id`s include a version in the format `-version-X` where X is
+an integer. For more information, see [ICB#517][].
+
+### Hashed References
+
+References prefixed by `hashed-` are defined as
+`first_16_chars(hex(sha256(utf8(chain_id))))`, with:
+
+- `chain_id`= the Tendermint `chain_id` from the genesis file (see above)
+- `utf8`= a UTF-8 encoding function
+- `sha256`= a SHA256 hash function
+- `hex`= a lowercase hex encoder
+- `first_16_chars`= a function to extract the first 16 characters of the
+  resulting string and dropping the rest
+
+### Resolution Method
+
+To resolve a blockchain reference for the Cosmos namespace, make a REST GET
+request to the blockchain node with endpoint `/node_info`, for example:
+
+```jsonc
+
+//TODO -- RPC ENDPOINT IS OUT OF DATE
+
+// Request
+curl -X GET "https://stargate.cosmos.network/node_info" -H "accept: application/json"
```

Comment:
```suggestion
// Request
curl -sS https://rpc.cosmos.network/status | jq .result.node_info
```

This will give you a different response because it is a different API now